extensions/amp-analytics/0.1/events.js

```diff
@@ -386,6 +405,84 @@ export class CustomEventTracker extends EventTracker {
   }
 }
 
+export class AmpStoryEventTracker extends CustomEventTracker {
+  /**
+   * @param {!./analytics-root.AnalyticsRoot} root
+   */
+  constructor(root) {
+    super(root);
+  }
+
+  /** @override */
+  add(context, eventType, config, listener) {
+    // TODO(artezan): add support for storySpec.
+    const rootTarget = this.root.getRootElement();
+
+    // Fire buffered events if any.
+    const buffer = this.buffer_ && this.buffer_[eventType];
+    if (buffer) {
+      const bufferLength = buffer.length;
+
+      for (let i = 0; i < bufferLength; i++) {
+        const event = buffer[i];
+        this.fireListener_(event, rootTarget, config, listener);
+      }
+    }
+
+    let observables = this.observables_[eventType];
+    if (!observables) {
+      observables = new Observable();
+      this.observables_[eventType] = observables;
+    }
+
+    return this.observables_[eventType].add(event => {
+      this.fireListener_(event, rootTarget, config, listener);
+    });
+  }
+
+  /**
+   * Fires listener given the specified configuration.
+   * @param {!AnalyticsEvent} event
+   * @param {!Element} rootTarget
+   * @param {!JsonObject} config
+
+   * @param {function(!AnalyticsEvent)} listener
+   */
+  fireListener_(event, rootTarget, config, listener) {
+    const type = event['type'];
+    const on = config['on'];
+
+    if (type !== on) {
+      return;
+    }
+
+    const vars = event['vars'];
+
+    listener(new AnalyticsEvent(rootTarget, type, vars));
+  }
+
+  /**
+   * Triggers a custom event for the associated root, or buffers them if the
+   * observables aren't present yet.
+   * @param {!AnalyticsEvent} event
+   */
+  trigger(event) {
+    const eventType = event['type'];
+    const observables = this.observables_[eventType];
+
+    // If listeners already present - trigger right away.
+    if (observables) {
+      observables.fire(event);
+    }
+
```

Comment:
sandboxEvent is added if an analytics event is added by an AMP component dynamically, in the format of `sandbox-xxx`. 
In those case, the event won't be recognized as a story event because the name doesn't start with `story-`. Would that be an issue? 
Code wise, it's fine extending the `CustomEventTracker` class, as no event would ever fall into the `this.sandboxBuffer_` here.

Reply:
That's right. All sandbox events will be handled by the `CustomEventTracker`. 
It won't be an issue for now, but it could be an issue if a component trigger a sandbox story event in the future. `sandbox-story-xxx`.  It would be great to keep that in mind by adding a comment.